projects/packages/publicize/src/class-publicize-setup.php

```diff
@@ -42,12 +42,21 @@ public static function on_jetpack_feature_publicize_enabled() {
 		add_action( 'rest_api_init', array( new Connections_Post_Field(), 'register_fields' ), 5 );
 		add_action( 'rest_api_init', array( new REST_Controller(), 'register_rest_routes' ) );
 		add_action( 'current_screen', array( static::class, 'init_sharing_limits' ) );
-		add_action( 'rest_api_init', array( new Jetpack_Social_Settings\Settings(), 'register_settings' ) );
-		add_action( 'admin_init', array( new Jetpack_Social_Settings\Settings(), 'register_settings' ) );
+
+		add_action( 'rest_api_init', array( static::class, 'register_core_options' ) );
+		add_action( 'admin_init', array( static::class, 'register_core_options' ) );
 
 		( new Social_Image_Generator\Setup() )->init();
 	}
 
+	/**
+	 * Registers the core options for the Publicize package.
+	 */
+	public static function register_core_options() {
```

Comment:
If `register_core_options` is the callback that should get called for both `register_core_options` for both `admin_init` and `rest_api_init` hooks, do we perhaps need just one hook?

Reply:
We discussed this on a call together, but just noting it here too,[ you need both](https://developer.wordpress.org/reference/functions/register_setting/#comment-3225) if you want it on the API too.